File: Reverse_Words/reverse_words.dart
class Solution {
  String reverseWords(String s) {
    final List<String> words = s.trim().split(RegExp(r'\s+'));

    int i = 0;
    int j = words.length - 1;

    while (i < j) {
      var temp = words[j];
      words[j] = words[i];
      words[i] = temp;
      i++;
      j--;
    }

    return words.join(" ");
  }
}

File: Reverse_Words/reverse_words.js
/**
 * @param {string} s - The string to reverse
 * @returns {string} returns words in reversal
 */
var reverseWords = function (s) {
	const words = s.trim().split(/\s+/);

	let i = 0;
	let j = words.length - 1;

	while (i < j) {
		[words[i], words[j]] = [words[j], words[i]];
		i++;
		j--;
	}

	return words.join(" ");
};
